extensions/amp-image-viewer/0.1/amp-image-viewer.js

```diff
@@ -119,61 +109,85 @@ export class ImageViewer {
     this.maxX_ = 0;
     /** @private {number} */
     this.maxY_ = 0;
-    /** @private {?./gesture.Gestures} */
+
+    /** @private {?../../../src/gesture.Gestures} */
     this.gestures_ = null;
-    /** @private {?function()} */
-    this.unlistenOnSwipePan_ = null;
 
-    /** @private {?./motion.Motion} */
+    /** @private {?../../../src/motion.Motion} */
     this.motion_ = null;
 
-    this.setupGestures_();
+    /** @private {?Element} */
+    this.sourceElement_ = null;
   }
 
-  /**
-   * Returns the root element of the image viewer.
-   * @return {!Element}
-   */
-  getElement() {
-    return this.viewer_;
+  /** @override */
+  buildCallback() {
+    this.vsync_ = this.getVsync();
+    this.element.classList.add('i-amphtml-image-lightbox-viewer');
+    const children = this.getRealChildren();
+    dev().assert(children.length == 1);
+    this.sourceElement_ = children[0];
+    this.setAsOwner(this.sourceElement_);
   }
 
-  /**
-   * Returns the img element of the image viewer.
-   * @return {!Element}
-   */
-  getImage() {
-    return this.image_;
+  /** @override */
+  layoutCallback() {
+    let loadPromise = Promise.resolve();
+    if (this.sourceElement_) {
+      this.scheduleLayout(this.sourceElement_);
+      loadPromise = this.sourceElement_.signals()
+          .whenSignal(CommonSignals.LOAD_END);
+    }
+    return loadPromise
+        .then(() => {
+          return this.vsync_.mutatePromise(() => {
+            if (!this.image_) {
+              this.image_ = this.win.document.createElement('img');
+              this.image_.classList.add('i-amphtml-image-viewer-image');
+
+              this.init_();
+              this.element.appendChild(this.image_);
+
+              this.element.removeChild(this.sourceElement_);
+              this.sourceElement_ = null;
+            }
+            this.setupGestures_();
+            this.measure();
+            this.updateSrc_();
+            this.registerOnResizeHandler_();
+          });
+        });
   }
 
-  /**
-   * Returns the boundaries of the viewer.
-   * @return {!./layout-rect.LayoutRectDef}
-   */
-  getViewerBox() {
-    return this.viewerBox_;
+  /** @override */
+  unlayoutCallback() {
+    this.gestures_.cleanup();
+    this.cleanupOnResizeHandler_();
+    return true;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.FILL;
```

Comment:
Do you have a standalone example of the usage for `<amp-lightbox-viewer>`? It would help determine what layouts should be supported.

Reply:
Nope. Removed that one because it's not a priority since lightbox is the only thing that we want to launch. I think Layout.FILL is the only thing we need for lightbox, and we can revisit this later.

Reply:
SGTM